test/functional/lua/uri_spec.lua

```diff
@@ -158,6 +158,18 @@ describe('URI methods', function()
 
   end)
 
+  describe('uri from bufnr', function()
+    it('Windows paths should not be treated as uris', function()
+      local file = 'C:\foo\bar.txt'
+      local uri = 'file:/C:/foo/bar.txt'
+        local test_case = string.format([[
+          local file = '%s'
+          return vim.uri_from_bufnr(vim.fn.bufadd(file))
+        ]], file)
+        eq(uri, exec_lua(test_case))
+    end)
+  end)
```

Comment:
This test will fail on non-Windows machines because in that case `vim.fn.bufadd` treats `'C:\foo\bar.txt'` as a literal file name in the current working directory. Unfortunately, I couldn't find a way to force vim to create a buffer with name `'C:\foo\bar.txt'`. Is there a way to restrict the test to run only on Windows hosts?

Reply:
You can use pending_win32 https://github.com/neovim/neovim/search?q=pending_win32&type=

Reply:
pending_win32 is only for skipping non-Windows tests. In this case, just guard it with condition that checks `helpers.iswin()`

Reply:
though we would not need to skip this if we just changed `uri_from_bufnr` to something like `uri_from_fname`. No idea why we have an API that takes a buffer number  just to call `nvim_buf_get_name` itself. It is less generic and saves only 1 line of code.

To unblock this PR and make this test less platform-sensitive, you can rename `uri_from_bufnr` to be "private" function name `_uri_from_fname`. Then `uri_from_bufnr` will just be a thin wrapper around that. Then the test can test `_uri_from_fname` instead of doing contortions to handle whatever OS it's running on.